backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt

```diff
@@ -847,8 +883,13 @@ internal class FunctionGenerationContext(val function: LLVMValueRef,
             }
         }
 
+        if (irClass.hasConstStateAndNoSideEffects(context)) {
+            return loadSlot(codegen.getObjectInstanceStorage(irClass, ObjectStorageKind.PERMANENT), false)
```

Comment:
What if the class has `@ThreadLocal` annotation?

Reply:
Don't think it makes sense to do have thread local storage in such a case.

Reply:
I mean that presence of `@ThreadLocal` is handled inconsistently here and in `LlvmDeclarations.kt`. The latter assumes that `@ThreadLocal object Foo {}` has thread-local instance field, while the former decides it is global.